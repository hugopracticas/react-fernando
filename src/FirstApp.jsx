import React from "react";
import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle }) => {
  const newMessage = {
    message: "Hola mundo",
    title: "Hugo",
  };

  const getName = (nombre) => {
    return `Hola ${nombre}`;
  };
  return (
    <div>
      <h1 data-testid="test-title">{getName("Hugo")}</h1>
      <h2>{title}</h2>
      <p data-testid="test-subtitle">{subTitle}</p>
      <p>{JSON.stringify(newMessage)}</p>
    </div>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  //title: "No hay titulo",
  subTitle: 1590,
};
