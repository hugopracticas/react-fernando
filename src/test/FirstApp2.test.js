import { getByText, render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";
describe("Pruebas en <FistApp2 />", () => {
  const title = "Hola, soy Goku";
  const subTitle = "Soy un subtitulo";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });
  test("debe de mostrar el mensaje Hola soy Hugo", () => {
    render(<FirstApp title={title} />);
    //screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe de hacer match con el titulo", () => {
    const { getByTestId } = render(<FirstApp title={title} />);
    expect(getByTestId("test-title").innerHTML).toBe("Hola Hugo");
  });

  test("debe de mostrar el titulo en un h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      "Hola Hugo"
    );
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    const { getByText, getByTestId } = render(
      <FirstApp title={title} subTitle={subTitle} />
    );

    expect(getByText(subTitle)).toBeTruthy();
    expect(getByTestId("test-subtitle").innerHTML).toBe(subTitle);
  });
});
