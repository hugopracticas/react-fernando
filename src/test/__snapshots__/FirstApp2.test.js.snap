// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FistApp2 /> debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
    >
      Hola Hugo
    </h1>
    <h2>
      Hola, soy Goku
    </h2>
    <p
      data-testid="test-subtitle"
    >
      1590
    </p>
    <p>
      {"message":"Hola mundo","title":"Hugo"}
    </p>
  </div>
</div>
`;
